src/DynamoCoreWpf/HelixRenderPackage.cs

```diff
@@ -0,0 +1,560 @@
+            // an end point, we duplicate every point except the first
+            // and the last.
+
+            lineStripVertexCounts.Add(n);
+
+            var idxCount = lines.Indices.Any() ? lines.Indices.Last() + 1 : 0;
+
+            for (var i = 0; i < n; ++i)
+            {
+                if (i != 0 && i != n - 1)
+                {
+                    lines.Indices.Add(idxCount);
+                }
+
+                lines.Indices.Add(idxCount);
+                idxCount++;
+            }
+        }
+
+        /// <summary>
+        /// Add a line strip vertex color to the render package.
+        /// </summary>
+        public void AddLineStripVertexColor(byte red, byte green, byte blue, byte alpha)
+        {
+            lines.Colors.Add(Color4FromBytes(red,green,blue,alpha));
+        }
+
+        /// <summary>
+        /// Apply a color to a sequence of point vertices.
+        /// </summary>
+        //public void ApplyPointVertexColors(int startIndex, int endIndex, byte red, byte green, byte blue, byte alpha)
+        //{
+        //    var message = string.Empty;
+        //    if (!HasValidStartEnd(startIndex, endIndex, points, out message))
+        //    {
+        //        if (!string.IsNullOrEmpty(message))
+        //        {
+        //            throw new Exception(message);
+        //        }
+
+        //        return;
+        //    }
+
+        //    for (var i = startIndex; i <= endIndex; i++)
+        //    {
+        //        points.Colors[i] = Color4FromBytes(red, green, blue, alpha);
+        //    }
+        //}
+
+        /// <summary>
+        /// Apply a color to each point vertex.
+        /// </summary>
+        /// <param name="colors">A buffer of R,G,B,A values corresponding to each vertex.</param>
+        public void ApplyPointVertexColors(byte[] colors)
+        {
+            if (colors.Count()/4 != points.Positions.Count)
+            {
+                throw new Exception("The number of colors specified must be equal to the number of vertices.");
+            }
+
+            points.Colors = null;
+            points.Colors = colors.ToColor4Collection();
+        }
+
+        /// <summary>
+        /// Apply a color to a sequence of line vertices.
+        /// </summary>
+        //public void ApplyLineVertexColors(int startIndex, int endIndex, byte red, byte green, byte blue, byte alpha)
+        //{
+        //    var message = string.Empty;
+        //    if (!HasValidStartEnd(startIndex, endIndex, lines, out message))
+        //    {
+        //        if (!string.IsNullOrEmpty(message))
+        //        {
+        //            throw new Exception(message);
+        //        }
+
+        //        return;
+        //    }
+
+        //    for (var i = startIndex; i <= endIndex; i++)
+        //    {
+        //        lines.Colors[i] = Color4FromBytes(red, green, blue, alpha);
+        //    }
+        //}
+
```

Comment:
Not entirely dead. I'm leaving this, as I need it for part of the solution to z fighting.


On May 4, 2015, at 1:42 PM, Peter Boyer <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/DynamoCoreWpf/HelixRenderPackage.cs<https://github.com/DynamoDS/Dynamo/pull/4291#discussion_r29621517>:

> +        //    if (!HasValidStartEnd(startIndex, endIndex, lines, out message))
> +        //    {
> +        //        if (!string.IsNullOrEmpty(message))
> +        //        {
> +        //            throw new Exception(message);
> +        //        }
> +
> +        //        return;
> +        //    }
> +
> +        //    for (var i = startIndex; i <= endIndex; i++)
> +        //    {
> +        //        lines.Colors[i] = Color4FromBytes(red, green, blue, alpha);
> +        //    }
> +        //}
> +


Event more dead code to remove

-
Reply to this email directly or view it on GitHub<https://github.com/DynamoDS/Dynamo/pull/4291/files#r29621517>.